examples/lock-app/silabs/src/LockManager.cpp

```diff
@@ -353,34 +333,78 @@ bool LockManager::Unbolt(chip::EndpointId endpointId, const Nullable<chip::Fabri
 
 bool LockManager::GetUser(chip::EndpointId endpointId, uint16_t userIndex, EmberAfPluginDoorLockUserInfo & user)
 {
+    CHIP_ERROR error;
+
     VerifyOrReturnValue(userIndex > 0, false); // indices are one-indexed
 
     userIndex--;
 
     VerifyOrReturnValue(IsValidUserIndex(userIndex), false);
 
+    VerifyOrReturnValue(kInvalidEndpointId != endpointId, false);
+
     ChipLogProgress(Zcl, "Door Lock App: LockManager::GetUser [endpoint=%d,userIndex=%hu]", endpointId, userIndex);
 
-    const auto & userInDb = mLockUsers[userIndex];
+    chip::StorageKeyName userKey = LockUserEndpoint(userIndex, endpointId);
+
+    uint16_t size = LockUserInfoSize;
 
-    user.userStatus = userInDb.userStatus;
-    if (UserStatusEnum::kAvailable == user.userStatus)
+    error = mStorage->SyncGetKeyValue(userKey.KeyName(), &mUserInStorage, size);
```

Comment:
Shouldn't we be checking that we actually got `LockUserInfoSize` bytes out?

Reply:
Added checks after each SyncGetKeyValue to check the size outparam